homeassistant/components/bouncie/sensor.py

```diff
@@ -0,0 +1,202 @@
+"""Bouncie Sensors."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import BouncieDataUpdateCoordinator, const
+
+ATTRIBUTION = "Data provided by Bouncie"
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Bouncie sensor entities based on a config entry."""
+    coordinator = hass.data[const.DOMAIN][config_entry.entry_id]
+    vehicles = coordinator.data["vehicles"]
+    sensor_types: list[SensorEntityDescription] = []
+    for vehicle in vehicles:
+        sensor_types.append(
+            SensorEntityDescription(
+                key=f"car-info-{vehicle[const.VEHICLE_VIN_KEY]}",
+                icon="mdi:car",
+                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]}",
+                state_class=SensorStateClass.MEASUREMENT,
+            )
+        )
+        sensor_types.append(
+            SensorEntityDescription(
+                key=f"car-odometer-{vehicle[const.VEHICLE_VIN_KEY]}",
+                icon="mdi:counter",
+                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} Odometer",
+                device_class="distance",
+            )
+        )
+        sensor_types.append(
+            SensorEntityDescription(
+                key=f"car-address-{vehicle[const.VEHICLE_VIN_KEY]}",
+                icon="mdi:map-marker",
+                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} Address",
+            )
+        )
+        sensor_types.append(
+            SensorEntityDescription(
+                key=f"car-fuel-{vehicle[const.VEHICLE_VIN_KEY]}",
+                icon="mdi:gas-station",
+                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} Fuel",
+                device_class="battery",
+            )
+        )
+        sensor_types.append(
+            SensorEntityDescription(
+                key=f"car-speed-{vehicle[const.VEHICLE_VIN_KEY]}",
+                icon="mdi:speedometer",
+                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} Speed",
+                device_class="speed",
```

Comment:
You should use constants only for device_class, "str" type will be removed soon, see #83391